sway/commands/output/non_desktop.c

```diff
@@ -0,0 +1,106 @@
+#include <strings.h>
+#include <wlr/types/wlr_output_damage.h>
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "sway/output.h"
+#include "sway/server.h"
+#include "log.h"
+#include "util.h"
+
+static int find_non_desktop_output(const void *item, const void *data)
+{
+	const struct sway_output_non_desktop *output = item;
+	const char *name = data;
+	return strcmp(output->wlr_output->name, name);
+}
+
+static int find_output(const void *item, const void *data)
+{
+	const struct sway_output *output = item;
+	const char *name = data;
+	return strcmp(output->wlr_output->name, name);
+}
+
+struct cmd_results *output_cmd_non_desktop(int argc, char **argv) {
+	if (argc == 0) {
+		return cmd_results_new(CMD_INVALID, "Missing non_desktop argument");
+	}
+
+	struct output_config *oc = config->handler_context.output_config;
+
+	if (strcmp(oc->name, "*") == 0) {
+		return cmd_results_new(CMD_INVALID,
+			"Cannot apply non_desktop to all outputs");
+	}
+
+	if (parse_boolean(argv[0], true)) {
+		int index = list_seq_find(root->outputs, find_output, oc->name);
+		if (index == -1) {
+			return cmd_results_new(CMD_FAILURE, "unknown output %s ", oc->name);
+		}
+
+		struct sway_output *output = root->outputs->items[index];
+		struct wlr_output *wlr_output = output->wlr_output;
+		wlr_output->non_desktop = true;
```

Comment:
Hmm. Do we really need to do this? I don't like writing to the `wlr_output` struct.

Reply:
I don't recally why I did that, maybe a leftover from a previous draft. I'll remove it.